### PR TITLE
[Doc] update  the English version of `custom_dataset.md`

### PR DESCRIPTION
**Pull request by tianleiSHI**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by PeterH0323** [comment]
Hi @tianleiSHI 
Thx for your contribute ! 
Pls using the latest dev branch to translate.  https://github.com/open-mmlab/mmyolo/blob/dev/docs/zh_cn/user_guides/custom_dataset.md

File: docs/en/user_guides/custom_dataset.md [hunk 1]
```diff
@@ -1,4 +1,1332 @@
...
+
+```{SeeAlso}
+The video of this document has been posted on Bilibili: [A nanny level tutorials for custom datasets from annotationt to deployment](https://www.bilibili.com/video/BV1RG4y137i5)
+```
+
+```{Note}
+All instructions in this document are done on Linux and are fully available on Windows, only slightly different in commands and operations.
+```
+
+Default that you have completed the installation of MMYOLO, if not installed, please refer to the document \[start your first step\] (https://mmyolo.readthedocs.io/zh_CN/latest/get_started.html#id1) for installation.
```
Comment (PeterH0323) [inline docs/en/user_guides/custom_dataset.md:13]:
```suggestion
Default that you have completed the installation of MMYOLO, if not installed, please refer to the document [GET STARTED](https://mmyolo.readthedocs.io/en/latest/get_started.html) for installation.
```

Comment (PeterH0323) [inline docs/en/user_guides/custom_dataset.md:13]:
```suggestion
Default that you have completed the installation of MMYOLO, if not installed, please refer to the document [GET STARTED](https://mmyolo.readthedocs.io/en/latest/get_started.html) for installation.
```

File: docs/en/user_guides/custom_dataset.md [hunk 2]
```diff
@@ -1,4 +1,1332 @@
...
+```shell
+.
+└── $OUT_DIR
+    ├── image1.json
+    ├── image1.json
+    └── ...
+```
+
+Here is an example of the original image and it's
+generating json file:
```
Comment (PeterH0323) [inline docs/en/user_guides/custom_dataset.md:169]:
```suggestion
Here is an example of the original image and it's generating json file:
```

File: docs/en/user_guides/custom_dataset.md [hunk 3]
```diff
@@ -1,4 +1,1332 @@
...
+
+<div align=center>
+  <img src="https://user-images.githubusercontent.com/25873202/205471430-dcc882dd-16bb-45e4-938f-6b62ab3dff19.jpg" alt="图片" width="45%"/>
+  <img src="https://user-images.githubusercontent.com/25873202/205471559-643aecc8-7fa3-4fff-be51-2fb0a570fdd3.png" alt="图片" width="45%"/>
```
Comment (PeterH0323) [inline docs/en/user_guides/custom_dataset.md:173]:
```suggestion
  <img src="https://user-images.githubusercontent.com/25873202/205471430-dcc882dd-16bb-45e4-938f-6b62ab3dff19.jpg" alt="Image" width="45%"/>
  <img src="https://user-images.githubusercontent.com/25873202/205471559-643aecc8-7fa3-4fff-be51-2fb0a570fdd3.png" alt="Image" width="45%"/>
```

File: docs/en/user_guides/custom_dataset.md (hunk 1 above)
Comment (PeterH0323) [reply docs/en/user_guides/custom_dataset.md:13]: 请将其余的中文链接也换为对应的英文文档链接

File: docs/en/user_guides/custom_dataset.md [hunk 4]
```diff
@@ -1,4 +1,1331 @@
 # The whole process of custom dataset annotation+training+testing+deployment
```
Comment (hhaAndroid) [inline docs/en/user_guides/custom_dataset.md:1]: 标题有点长，可以换一个名字感觉

Comment (tianleiSHI) [reply docs/en/user_guides/custom_dataset.md]: okey

File: docs/en/user_guides/custom_dataset.md [hunk 5]
```diff
@@ -1,4 +1,1331 @@
...
+
+```shell
+python demo/image_demo.py ./data/cat/images \
+                          ./configs/custom_dataset/yolov6_s_syncbn_fast_1xb32-100e_cat.py \
+                          ./work_dirs/yolov6_s_syncbn_fast_1xb32-100e_cat/best_coco/bbox_mAP_epoch_96.pth \
+                          --out-dir ./data/cat/pred_images
+```
+
+<div align=center>
+<img src="https://user-images.githubusercontent.com/25873202/204773727-5d3cbbad-1265-45a0-822a-887713555049.jpg" alt="推理图片"/>
```
Comment (PeterH0323) [inline docs/en/user_guides/custom_dataset.md:937]:
```suggestion
<img src="https://user-images.githubusercontent.com/25873202/204773727-5d3cbbad-1265-45a0-822a-887713555049.jpg" alt="Image"/>
```

File: docs/en/user_guides/custom_dataset.md [hunk 6]
```diff
@@ -1,4 +1,1331 @@
...
+Then install it from the command line:
+
+```shell
+pip install wandb
+# After running wandb login, enter the API Keys obtained above, and the login is successful.
+wandb login
+```
+
+<div align=center>
+<img src="https://user-images.githubusercontent.com/25873202/206070473-201795e0-c81f-4247-842a-16d6acae0474.png" alt="推理图片"/>
```
Comment (PeterH0323) [inline docs/en/user_guides/custom_dataset.md:659]:
```suggestion
<img src="https://user-images.githubusercontent.com/25873202/206070473-201795e0-c81f-4247-842a-16d6acae0474.png" alt="Image"/>
```

File: docs/en/user_guides/custom_dataset.md [hunk 7]
```diff
@@ -1,4 +1,1331 @@
...
+anchors = [  # the anchor has been updated according to the characteristics of dataset. The generation of anchor will be explained in the following section.
+    [(68, 69), (154, 91), (143, 162)],  # P3/8
+    [(242, 160), (189, 287), (391, 207)],  # P4/16
+    [(353, 337), (539, 341), (443, 432)]  # P5/32
+]
+
+class_name = ('cat', )  # according to the label information of class_with_id.txt, set the class_name
+num_classes = len(class_name)
+metainfo = dict(
+    CLASSES=class_name,
```
Comment (hhaAndroid) [inline docs/en/user_guides/custom_dataset.md:403]:
```suggestion
    classes=class_name,
```

File: docs/en/user_guides/custom_dataset.md [hunk 8]
```diff
@@ -1,4 +1,1331 @@
...
+    PALETTE=[(220, 20, 60)]  # the color of drawing, free to set
```
Comment (hhaAndroid) [inline docs/en/user_guides/custom_dataset.md:404]:
```suggestion
    palette=[(220, 20, 60)]  # the color of drawing, free to set
```

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**